### PR TITLE
Migrate route.sh, run.sh, service.sh and status.sh tests to go

### PR DESCRIPTION
**Pull request by ardaguclu**
This PR migrates routes.sh, run.sh, service.sh and status.sh tests to go tests in order to enable various platforms can run these tests.

### CODE REVIEW

**Comment by ardaguclu** [comment]
Route tests might have impact on different network stacks like;

/test e2e-metal-ipi
/test e2e-metal-ipi-ovn-dualstack

**Comment by ardaguclu** [comment]
All tests introduced by this PR passed in all variants without any problem.

File: test/extended/cli/routes.go [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+		err = oc.Run("create").Args("route", "edge", "new-route").Execute()
+		o.Expect(err).To(o.HaveOccurred())
+
+		err = oc.Run("delete").Args("services", "frontend").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = oc.Run("create").Args("route", "edge", "--insecure-policy=Allow", "--service=foo", "--port=80").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		out, err = oc.Run("get").Args("route", "foo", "-o", "jsonpath=\"{.spec.tls.insecureEdgeTerminationPolicy}\"").Output()
```
Comment (soltysh) [inline test/extended/cli/routes.go:69]: Nit: you can use backticks to not to have to escape double quote.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Comment by ardaguclu** [comment]
This PR has no urgency and I'd prefer waiting rebase first to not block it.

/hold

**Comment by soltysh** [comment]
> This PR has no urgency and I'd prefer waiting rebase first to not block it.

k8s bump is temporarily blocked waiting for other PRs, so I'm fine with this landing
/hold cancel
/retest-required

**Comment by soltysh** [comment]
Infra failure on fips job, but other e2e passed, which is sufficient signal that test are working as expected.
/override ci/prow/e2e-aws-ovn-fips

**Comment by soltysh** [comment]
Unrelated failure to test changes, and this will actually improve the overall state of tests. 
/override ci/prow/e2e-gcp-ovn-upgrade

**Pull request merged**